notebooks/working/ho_deconfounder_on_sim_data.py

```diff
@@ -0,0 +1,811 @@
+
+mnl_specification["total_travel_distance"] = [1, 2, 3, 7, 8]
+mnl_names["total_travel_distance"] = ['Travel Distance, units:mi (Drive Alone)',
+                                      'Travel Distance, units:mi (SharedRide-2)',
+                                      'Travel Distance, units:mi (SharedRide-3+)',
+                                      'Travel Distance, units:mi (Walk)',
+                                      'Travel Distance, units:mi (Bike)']
+
+# mnl_specification["cross_bay"] = [[2, 3], [4, 5, 6]]
+# mnl_names["cross_bay"] = ["Cross-Bay Tour (Shared Ride 2 & 3+)",
+#                           "Cross-Bay Tour (All Transit Modes)"]
+mnl_specification["cross_bay"] = [[2, 3]]
+mnl_names["cross_bay"] = ["Cross-Bay Tour (Shared Ride 2 & 3+)"]
+
+mnl_specification["household_size"] = [[2, 3]]
+mnl_names["household_size"] = ['Household Size (Shared Ride 2 & 3+)']
+
+mnl_specification["num_kids"] = [[2, 3]]
+mnl_names["num_kids"] = ["Number of Kids in Household (Shared Ride 2 & 3+)"]
+# -
+
+# # Deconfounder
+
+# +
+import tensorflow as tf
+import numpy as np
+import numpy.random as npr
+import pandas as pd
+import tensorflow as tf
+import tensorflow_probability as tfp
+import statsmodels.api as sm
+
+from tensorflow_probability import edward2 as ed
+from sklearn.datasets import load_breast_cancer
+from pandas.plotting import scatter_matrix
+from scipy import sparse, stats
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import LogisticRegression 
+from sklearn.metrics import confusion_matrix, classification_report, roc_auc_score, roc_curve
+
+import matplotlib
+matplotlib.rcParams.update({'font.sans-serif' : 'Helvetica',
+                            'axes.labelsize': 10,
+                            'xtick.labelsize' : 6,
+                            'ytick.labelsize' : 6,
+                            'axes.titlesize' : 10})
+import matplotlib.pyplot as plt
+
+import seaborn as sns
+color_names = ["windows blue",
+               "amber",
+               "crimson",
+               "faded green",
+               "dusty purple",
+               "greyish"]
+colors = sns.xkcd_palette(color_names)
+sns.set(style="white", palette=sns.xkcd_palette(color_names), color_codes = False)
+
+# +
+X_columns = ['total_travel_time',
+       'total_travel_cost', 
+#              'total_travel_distance', 
+             'cross_bay', 'household_size', 'num_kids', 
+              'cars_per_licensed_drivers', 
+             'gender'
+             
+            ]
+
+y_column = data['mode_id']
+# -
+
+spec_dic = specifications(mnl_specification=mnl_specification, num_modes=8)
+spec_dic
+
+confounder_vectors = []
+holdout_dfs = []
+masks_df = []
+rows_df =[]
+latent_dim = 1
+for i in data['mode_id'].unique():
+
+    data_mode_i = data[data['mode_id']==i]
+    # standardize the data for PPCA
+    print("Analysis for mode: ", i)
+    print("-------------------------------------------------------------------------------------------")
```

Comment:
The following should accomplish the same thing without taking up multiple lines. Feel free to use whatever number you want, 40 just seemed like a big enough value.
```suggestion
    print("-" * 40)
```